File: Easy/2418_Sort_the_People/Solution.java



public class Solution {
    public String[] sortPeople(String[] names, int[] heights) {
        for (int i = 0; i < heights.length; i++) {
            int maxIdx = i;

            for (int j = i; j < heights.length; j++) {
                if (heights[maxIdx] < heights[j]) 
                    maxIdx = j;
            }

            int tempHeight = heights[i];
            heights[i] = heights[maxIdx];
            heights[maxIdx] = tempHeight;

            String tempName = names[i];
            names[i] = names[maxIdx];
            names[maxIdx] = tempName;
            
        }

        return names;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        String[] names1 = {"Mary", "John", "Emma"};
        int[] heights1 = {180, 165, 170};
        String[] result1 = solution.sortPeople(names1, heights1);
        for (String name : result1) {
            System.out.print(name + " ");
        }
        System.out.println();
        
        // test cases 2
        String[] names2 = {"Alice", "Bob", "Bob"};
        int[] heights2 = {155, 185, 150};
        String[] result2 = solution.sortPeople(names2, heights2);
        for (String name : result2) {
            System.out.print(name + " ");
        }
        System.out.println();
        
    }
}

File: Easy/2418_Sort_the_People/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        for (int i = 0; i < heights.size(); i++) {
            int maxIdx = i;

            for (int j = i; j < heights.size(); j++) {
                if (heights[maxIdx] < heights[j]) 
                    maxIdx = j;
            }

            swap(heights[i], heights[maxIdx]);
            swap(names[i], names[maxIdx]);
        }

        return names;
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    vector<string> names1 = {"Mary","John","Emma"};
    vector<int> heights1 = {180,165,170};
    vector<string> result1 = solution.sortPeople(names1, heights1);
    printVec(result1);

    // test cases 2
    vector<string> names2 = {"Alice","Bob","Bob"};
    vector<int> heights2 = {155,185,150};
    vector<string> result2 = solution.sortPeople(names2, heights2);
    printVec(result2);

    return 0;
}